src/utils/mod.rs

```diff
@@ -177,48 +159,55 @@ impl Default for Tpm2BPublicBuilder {
 #[allow(missing_debug_implementations)]
 #[derive(Copy, Clone, Default)]
 pub struct TpmsRsaParmsBuilder {
-    symmetric: TPMT_SYM_DEF_OBJECT,
-    scheme: Option<AsymSchemeUnion>,
-    key_bits: TPMI_RSA_KEY_BITS,
-    exponent: u32,
+    /// Symmetric cipher to be used in conjuction with the key
+    pub symmetric: Option<TPMT_SYM_DEF_OBJECT>,
+    /// Asymmetric scheme to be used for key operations
+    pub scheme: Option<AsymSchemeUnion>,
+    /// Size of key, in bits
+    pub key_bits: TPMI_RSA_KEY_BITS,
+    /// Public exponent of the key. A value of 0 defaults to 2 ^ 16 + 1
+    pub exponent: u32,
+    /// Flag indicating whether the key shall be used for signing
+    pub for_signing: bool,
+    /// Flag indicating whether the key shall be used for decryption
+    pub for_decryption: bool,
+    /// Flag indicating whether the key is restricted
+    pub restricted: bool,
```

Comment:
Could those fields be private so that we can control that only builders with valid parameters are created?

Reply:
The thing about builders is that you can't check that they're valid at point X in time. The only time where it makes sense to check is when `build` is called. Otherwise you have to do the same checks every time when a new parameter is added and there might be things that are impossible to do because you'd need to change two parameters simultaneously to make the builder valid.